controllers/che/backup.go

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package che
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	semver "github.com/blang/semver/v4"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	chev1 "github.com/eclipse-che/che-operator/api/v1"
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+)
+
+const (
+	DefaultBackupServerConfigLabelKey = "che.eclipse.org/default-backup-server-configuration"
+)
+
+func isCheGoingToBeUpdated(cheCR *chev1.CheCluster) bool {
+	deployedCheVersion := cheCR.Status.CheVersion
+	newCheVersion := deploy.DefaultCheVersion()
+	if deployedCheVersion == "" || deployedCheVersion == "next" || deployedCheVersion == newCheVersion {
+		return false
+	}
+
+	deployedSemver, err := semver.Make(deployedCheVersion)
+	if err != nil {
+		logrus.Warn(getSemverParseErrorMessage(deployedCheVersion, err))
+		return false
+	}
+	currentSemver, err := semver.Make(newCheVersion)
+	if err != nil {
+		logrus.Warn(getSemverParseErrorMessage(newCheVersion, err))
+		return false
+	}
+	return currentSemver.GT(deployedSemver)
+}
+
+func getSemverParseErrorMessage(version string, err error) string {
+	return fmt.Sprintf("It is not possible to parse a current version '%s'. Cause: %v", version, err)
+}
+
+func getBackupCR(deployContext *deploy.DeployContext) (*chev1.CheClusterBackup, error) {
+	backupCR := &chev1.CheClusterBackup{}
+	backupCRName := getBackupCRNameForVersion(deploy.DefaultCheVersion())
+	backupCRNamespacedName := types.NamespacedName{Namespace: deployContext.CheCluster.GetNamespace(), Name: backupCRName}
+	if err := deployContext.ClusterAPI.Client.Get(context.TODO(), backupCRNamespacedName, backupCR); err != nil {
+		return nil, err
+	}
+	return backupCR, nil
+}
+
+func getBackupCRNameForVersion(version string) string {
+	return "backup-before-update-to-" + strings.Replace(version, ".", "-", -1)
+}
+
+func requestNewBackup(deployContext *deploy.DeployContext) error {
+	backupCR, err := getBackupCRSpec(deployContext)
+	if err != nil {
+		return err
+	}
+	return deployContext.ClusterAPI.Client.Create(context.TODO(), backupCR)
+}
+
+func getBackupCRSpec(deployContext *deploy.DeployContext) (*chev1.CheClusterBackup, error) {
+	backupServerConfigName, err := getDefaultBackupServer(deployContext)
+	if err != nil {
+		return nil, err
+	}
+	cheClusterBackupSpec := chev1.CheClusterBackupSpec{}
+	if backupServerConfigName == "" {
+		cheClusterBackupSpec.UseInternalBackupServer = true
+	} else {
+		cheClusterBackupSpec.UseInternalBackupServer = false
```

Comment:
Is it really needed to set it `false` when `BackupServerConfigRef` is set?

Reply:
No, just for clarity.